### ISSUE TITLE
deviceready is not fired if I specify Hostname preference

### ISSUE THREAD

**Issue by BobDankert** [ionic-team/cordova-plugin-ionic-webview#432]
I've been dealing with an issue where the deviceready does not fire in my app in Android. I've tried removing plugins, adding plugins, remove/add platforms, etc. After various tests, I've been able to narrow this down to an issue where deviceready does not fire on my device when I specify a Hostname in the config.xml of my app. As soon as I remove this line then the deviceready fires without issue. I have allow-intent that I believe covers my new hostname, so I don't think it's related to that:
```
<allow-intent href="http://*/*" />
<allow-intent href="https://*/*" />
```

I'm running:
```
Cordova 9.0.0
Ionic CLI: 5.2.3
cordova-android 8.0.0
```

Plugins:
```
PS >> cordova plugins ls
cordova-background-geolocation 3.2.2 "BackgroundGeolocation"
cordova-plugin-background-fetch 5.6.0 "CDVBackgroundFetch"
cordova-plugin-bluetooth-serial 0.4.7 "Bluetooth Serial"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-geolocation 4.0.2 "Geolocation"
cordova-plugin-ionic-keyboard 2.1.3 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 4.1.1 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
cordova-support-google-services 1.1.0 "cordova-support-google-services"
phonegap-plugin-multidex 1.0.0 "Multidex"
phonegap-plugin-push 2.2.3 "PushPlugin"
```

If there is any other information you need, let me know and I will do my best to get it.

**Comment by jcesarmobile** [comment]
As explained in the hostname preference section https://github.com/ionic-team/cordova-plugin-ionic-webview#hostname

> If you change it, you'll need to add a new allow-navigation entry in the config.xml for the configured url (i.e <allow-navigation href="http://app/*"/> if Hostname is set to app). This is only needed for the Android url when using http://, https:// or a custom scheme. All ionic:// urls are whitelisted by the plugin.

**Comment by BobDankert** [comment]
Thank you - I was confusing allow-intent and allow-navigation. Can confirm that this is working as expected with the proper allow-navigation in the config.xml.

**Comment by tymcdowell** [comment]
If I set the the Hostname preference, the deviceready event never fires.  I have added the allow-navigation tags to match.  CORS setup is fine and matches.  Can see xhr requests going through with new hostname.  If I remove the hostname, then all works fine.

Config below.  More allow-navigation entries than needed.  Just trying everything. 

    <preference name="Hostname" value="myapp" />
    <allow-navigation href="*" launch-external="yes" />
    <allow-navigation href="http://myapp/*" />
    <allow-navigation href="https://myapp/*" />
    <allow-navigation href="ionic://myapp*" />
    <allow-navigation href="ionic://myapp/*" />
    <allow-navigation href="ionic://localhost" />
    <allow-navigation href="http://localhost" />
    <allow-navigation href="https://localhost" />

Installed platforms:
  ios 5.1.1

Plugins:
branch-cordova-sdk 4.1.2 "branch-cordova-sdk"
cordova-plugin-actionsheet 2.3.3 "ActionSheet"
cordova-plugin-android-permissions 1.0.2 "Permissions"
cordova-plugin-appavailability 0.4.2 "AppAvailability"
cordova-plugin-appinfo 2.1.2 "AppInfo Plugin"
cordova-plugin-backbutton 0.3.0 "Backbutton"
cordova-plugin-calendar 5.1.5 "Calendar"
cordova-plugin-camera 4.1.0 "Camera"
cordova-plugin-device 2.0.3 "Device"
cordova-plugin-dialogs 2.0.2 "Notification"
cordova-plugin-file 6.0.2 "File"
cordova-plugin-file-transfer 1.7.1 "File Transfer"
cordova-plugin-geolocation 4.0.2 "Geolocation"
cordova-plugin-inappbrowser 3.2.0 "InAppBrowser"
cordova-plugin-ionic-webview 4.1.3 "cordova-plugin-ionic-webview"
cordova-plugin-media 5.0.3 "Media"
cordova-plugin-media-capture 3.0.3 "Capture"
cordova-plugin-network-information 2.0.2 "Network Information"
cordova-plugin-safariviewcontroller 1.6.0 "SafariViewController"
cordova-plugin-spinner 1.1.0 "Spinner Plugin"
cordova-plugin-splashscreen 5.0.3 "Splashscreen"
cordova-plugin-vibration 3.1.1 "Vibration"
cordova-plugin-whitelist 1.3.4 "Whitelist"
cordova-plugin-x-socialsharing 5.6.4 "SocialSharing"
es6-promise-plugin 4.2.2 "Promise"
ionic-plugin-keyboard 2.2.1 "Keyboard"
org.devgeeks.Canvas2ImagePlugin 0.6.0 "Canvas 2 Image"
urbanairship-cordova 8.0.0 "Urban Airship"
